app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt

```diff
@@ -108,11 +111,13 @@ private fun Background(
 @Destination<RootGraph>
 @Composable
 fun AnimatedVisibilityScope.ReaderScreen(args: ReaderScreenArgs, navigator: DestinationsNavigator) = composing(navigator) {
-    Await({ preparePageLoader(args) }) { pageLoader ->
-        DisposableEffect(pageLoader) {
-            pageLoader.start()
-            onDispose {
-                pageLoader.stop()
+    Await(with(asyncInVM { preparePageLoader(args) }) { { await() } }) { loader ->
+        launchInVM {
+            loader.start()
+            try {
+                awaitCancellation()
+            } finally {
+                loader.stop()
```

Comment:
If the use opens an external archive, the old archive in reader won't be replaced.

Reply:
We need to cancel & relunch when `args` changes.

Reply:
Done